packages/babel-plugin/src/babel-plugin.ts

```diff
@@ -80,7 +86,9 @@ export default declare<State>((api) => {
               const jsxSourceMatches = JSX_SOURCE_ANNOTATION_REGEX.exec(comment.value);
               const jsxMatches = JSX_ANNOTATION_REGEX.exec(comment.value);
 
-              if (jsxSourceMatches && jsxSourceMatches[1] === COMPILED_MODULE) {
+              // jsxPragmas currently only run on the top-level compiled module,
+              // hence we don't interrogate this.compiledModuleOrigins.
+              if (jsxSourceMatches && jsxSourceMatches[1] === DEFAULT_COMPILED_MODULE) {
```

Comment:
Call out: jsx pragma doesn't support custom modules. I didn't think it was worth the effort as it's annoying to set up as a consumer.